.github/workflows/build-emsdk.yml

```diff
@@ -1,27 +1,77 @@
+# any other branch. Also, skip any non-source changes from running on CI
+on:
+  push:
+    branches: main
+    paths-ignore:
+      - 'docs/**'
+      - 'examples/**'
+      - '.gitignore'
+      - '*.rst'
+      - '*.md'
+      - '.github/workflows/*.yml'
+      # re-include current file to not be excluded
+      - '!.github/workflows/build-emsdk.yml'
+
+  pull_request:
+    branches: main
+    paths-ignore:
+      - 'docs/**'
+      - 'examples/**'
+      - '.gitignore'
+      - '*.rst'
+      - '*.md'
+      - '.github/workflows/*.yml'
+      # re-include current file to not be excluded
+      - '!.github/workflows/build-emsdk.yml'
 
 jobs:
   build:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-22.04
+    env:
+      # pin SDK version to the latest, update manually
+      SDK_VERSION: 0.2.0
+      SDK_ARCHIVE: python-wasm-sdk-stable.tar.bz2
+
+      # use the most recent cython from github, but pin on a commit for CI
+      # stability. This is also needed to benefit from caching cython installs
+      LATEST_CYTHON_COMMIT: 8e29b6d47f6f5b10ec1a37f06db440156ac2ac2e
+
     steps:
     - uses: actions/checkout@v3.0.2
-    - name: pygame-wasm-builder prepare
+    
+    - name: Cache pip directory for cython
+      uses: actions/cache@v3
+      with:
+        path: ~/.cache/pip
+        key: wasm-ubuntu-cython-${{ env.LATEST_CYTHON_COMMIT }}
+
+    - name: Install latest cython and regen
+      run: |
+        pip install git+https://github.com/cython/cython.git@${{ env.LATEST_CYTHON_COMMIT }}
+        touch $(find | grep pxd$)
+        python3 setup.py cython_only
+
+    - name: Install python-wasm-sdk
+      run: |
+        mkdir python-wasm-sdk && cd python-wasm-sdk
+        curl -sL --retry 5 https://github.com/pygame-web/python-wasm-sdk/releases/download/$SDK_VERSION/$SDK_ARCHIVE | tar xvPj
+      working-directory: /opt
+
+    - name: Build WASM with emsdk
+      run: |
+        EMCC_CFLAGS="-I/opt/python-wasm-sdk/devices/emsdk/usr/include/SDL2 -s USE_SDL=2" /opt/python-wasm-sdk/python3-wasm setup.py build
+    
+    - name: Generate libpygame.a static binaries archive
       run: |
-            WD=$(pwd)
-            python3 -V
-            echo $WD
-            clang --version | head -n 1
-            echo
-            grep "^Pkg.Revision =" ${ANDROID_HOME}/ndk-bundle/source.properties
-            sudo apt-get update
-            sudo apt-get install bash wget
-            cd $GITHUB_WORKSPACE/..
-            git clone https://github.com/pmp-p/python-wasm-plus.git
-            mkdir $GITHUB_WORKSPACE/../python-wasm-plus/src
-            ln -s $(pwd)/pygame $GITHUB_WORKSPACE/../python-wasm-plus/src/pygame-wasm
-            cd python-wasm-plus
-            bash ./python-wasm-plus.sh
-            bash ./buildapp.sh
+        mkdir -p dist && cd dist
+        SYS_PYTHON=python3 /opt/python-wasm-sdk/emsdk/upstream/emscripten/emar rcs libpygame.a $(find build/lib.wasm32-*/ | grep o$)
```

Comment:
i think the working is not ok because of previous "cd dist" try `SYS_PYTHON=python3 /opt/python-wasm-sdk/emsdk/upstream/emscripten/emar rcs libpygame.a $(find $GITHUB_WORKSPACE/build/temp.wasm32-* | grep o$)`